### ISSUE TITLE
Remove use of UNSAFE_componentWillReceiveProps

### ISSUE THREAD

**Issue by Sventenhaaf** [iddan/react-spreadsheet#142]
In my Vitejs app, it complains that you're using legacy methods = UNSAFE_componentWillReceiveProps.

```
react-dom.development.js:67 Warning: Legacy context API has been detected within a strict-mode tree.

The old API will be supported in all 16.x releases, but applications using it should migrate to the new version.

Please update the following components: c, r2
```

I'm using a `<React.StrictMode>` wrapper when I see this error

Is it on the agenda to update these components?

**Comment by PCOffline** [comment]
Bumping this

**Comment by iddan** [comment]
Yes, I just haven’t gone around this. PRs are welcome

**Comment by PCOffline** [comment]
I can't even find the use of this method in your code..

**Comment by PCOffline** [comment]
Alright, did a little research. You use the `unistore `package. This package is **very** outdated. It uses a ton of legacy API and unsafe methods. You had another issue back in June 2020 about legacy Context API, which was presumably solved, but I still get errors about it.

The best solution would be to completely remove any use of that package from your code.

**Comment by iddan** [comment]
I agree unistore should be replaced. I'm not sure you must replace it to solve this particular issue. It seems like [React Context Selectors](https://github.com/reactjs/rfcs/pull/119) may come soon to React which will make remove the need for a state library.

**Comment by PCOffline** [comment]
Honestly, no clue. That post is from 2 years ago, and no one can promise React will focus on that all of a sudden.

 In my opinion, unistore is an outdated package. Their last commit was about two months ago and their last publish a year ago. Despite the many issues posted on their repository, I don't see any moves towards updating the package.

**Comment by iddan** [comment]
If you look at the end of the page @acdlite made progress with this about a month ago

**Comment by PCOffline** [comment]
Will the unistore package need to update accordingly if that feature will be released?

**Comment by iddan** [comment]
Ones the feature is out there would be no need of unistore. The `<Spreadsheet />` component will expose context of `data` and the `<Cell />` component will consume the specific data it needs.

**Comment by PCOffline** [comment]
I see. In my opinion there should be at least a temporary fix until that day comes, but it's your library after all, so you get to decide :)

**Comment by iddan** [comment]
I don’t mind to review and and merge a temporary fix but I won’t implement it myself.

**Comment by PCOffline** [comment]
Alright, I'm not very familiar with all the context systems, so I suppose we'll have to wait for the RFC to be released.

**Comment by PCOffline** [comment]
Though I will try to research and solve the issue in a non-breaking matter in my free time

**Comment by iddan** [comment]
I managed to migrate `Spreadsheet` and `SpreadsheetStateProvider` to hooks. I hope it will help a little if you get into that.

**Comment by iddan** [comment]
Should be solved with #166

**Comment by PCOffline** [comment]
Great to hear!

**Comment by iddan** [comment]
Should be solved in `react-spreadsheet@0.5.15`
